airflow/models/pool.py

```diff
@@ -91,7 +96,15 @@ def slots_stats(session: Session = None) -> Dict[str, PoolStats]:
 
         pools: Dict[str, PoolStats] = {}
 
-        pool_rows: Iterable[Tuple[str, int]] = session.query(Pool.pool, Pool.slots).all()
+        query = session.query(Pool.pool, Pool.slots)
```

Comment:
Those parameter names and the way the if's here are done are pretty confusing. "with_for_update" can be . Could we make the code a bit more readable and make it clear why we need for update in one place, but we do not need it in another? I think this all can be achieved by a single bool ("use_for_update") parameter ?

Reply:
Yeah, this went through a few iterations, and I agree it should be tided up now.

Reply:
Done in https://github.com/apache/airflow/pull/10956/commits/8f357fdf4630df47d4639706516733e0587480dd